packages/contracts-bedrock/src/L2/SuperchainWETH.sol

```diff
@@ -98,4 +117,48 @@ contract SuperchainWETH is WETH98, IERC7802, ISemver {
         return _interfaceId == type(IERC7802).interfaceId || _interfaceId == type(IERC20).interfaceId
             || _interfaceId == type(IERC165).interfaceId;
     }
+
+    /// @notice Sends ETH to some target address on another chain.
+    /// @param _to       Address to send tokens to.
+    /// @param _chainId  Chain ID of the destination chain.
+    function sendETH(address _to, uint256 _chainId) external payable returns (bytes32 msgHash_) {
+        if (_to == address(0)) revert ZeroAddress();
+
+        if (IL1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) {
```

Comment:
Probably not worth the complexity, but given https://github.com/ethereum-optimism/optimism/pull/12710/files#r1850695219, there exists a world where on a custom gas token chain it would just forward as weth rather than reverting. Then the ux for sending from/to eth/custom gas token would be the same and the user gets native asset or weth depending on the type of chain but same api

Reply:
I think `sendETH` is trickier to implement on a cgt chain because `msg.value` is not `ETH`, so wrapping to `SuperchainWETH` would not really be possible, right? is there a way you were thinking to work around this?